runtime-rs/dragonball: add support building kernel with upcall and GPU hotplug (#9244)
kata-containers/kata-containers: docs/use-cases/NVIDIA-GPU-passthrough-and-Kata.md

$ sudo -E ./build-kernel.sh -v 5.15.23 -g nvidia install
```

However, there are special cases like `Dragonball VMM`. It directly supports device hot-plug/hot-unplug
via upcall to avoid `ACPI` virtualization and minimize `VM` overhead. Since upcall isn't upstream kernel
code, using `Dragonball VMM` for NVIDIA GPU hot-plug/hot-unplug requires applying the Upcall patchset in

> addition to the above kernel configuration items. Therefore, follow the steps below to build for NVIDIA
> GPU hot-plug/hot-unplug in the `runtime-rs/Dragonball` scenario:

Reviewer: ```suggestion
addition to the above kernel configuration items. Follow these steps to build a kernel that supports NVIDIA GPU hot-plug/hot-unplug for `Dragonball`:
```

Author: Thx @justxuewei fixed

```sh 
# Prepare .config to support both upcall and nvidia gpu 
$ ./build-kernel.sh -v 5.10.25 -e -t dragonball -g nvidia -f setup

# Build guest kernel to support both upcall and nvidia gpu 
$ ./build-kernel.sh -v 5.10.25 -e -t dragonball -g nvidia build

# Install guest kernel to support both upcall and nvidia gpu
$ sudo -E ./build-kernel.sh -v 5.10.25 -e -t dragonball -g nvidia install
```

> **Note**:

> > `-v 5.10.25`: Specify the guest kernel version. 
> > `-e` here means experimental, mainly because `upcall` patches are not in upstream Linux kernel.
> > `-t dragonball` is for specifying hypervisor type
> > `-f` is for generating `.config` file

Reviewer: ```suggestion
> - `-v 5.10.25`: Specify the guest kernel version. 
> - `-e` here means experimental, mainly because `upcall` patches are not in upstream Linux kernel.
> - `-t dragonball` is for specifying hypervisor type
> - `-f` is for generating `.config` file
```

Author: fixed
